atomic_reactor/utils/koji.py

```diff
@@ -576,39 +589,38 @@ def get_repositories_and_digests(workflow, pullspec_image):
     """
     Returns a map of images to their repositories and a map of media types to each digest
 
-    it creates a map of images to digests, which is need to create the image->repository
+    it creates a map of images to digests, which is needed to create the image->repository
     map and uses the same loop structure as media_types->digest, but the image->digest
     map isn't needed after we have the image->repository map and can be discarded.
     """
     digests = {}  # image -> digests
     typed_digests = {}  # media_type -> digests
-    for registry in workflow.data.push_conf.docker_registries:
-        for image in workflow.data.tag_conf.images:
-            image_str = image.to_str()
-            if image_str in registry.digests:
-                image_digests = registry.digests[image_str]
-                digest_list = [select_digest(image_digests)]
-                digests[image.to_str(registry=False)] = digest_list
-                for digest_version in image_digests.content_type:
-                    if digest_version == 'v1':
-                        continue
-                    if digest_version not in image_digests:
-                        continue
-                    digest_type = get_manifest_media_type(digest_version)
-                    typed_digests[digest_type] = image_digests[digest_version]
-
-    registries = workflow.data.push_conf.all_registries
+
+    for image in workflow.data.tag_conf.images:
+        image_digests = get_manifest_digests(image, workflow.conf.registry['uri'],
+                                             workflow.conf.registry['insecure'],
+                                             workflow.conf.registry.get('secret', None))
+        if image_digests:
+            digest_list = [select_digest(image_digests)]
+            digests[image.to_str(registry=False)] = digest_list
+            for digest_version in image_digests.content_type:
+                if digest_version == 'v1':
+                    continue
+                if digest_version not in image_digests:
+                    continue
+                digest_type = get_manifest_media_type(digest_version)
+                typed_digests[digest_type] = image_digests[digest_version]
```

Comment:
Do we actually need to get the manifest digests for all the images in tag_conf? Couldn't we just get them for the pullspec_image?

This whole method makes very little sense to me, but I *think* the only reason for processing every possible registry+image combination was that the same image in two different registries may have different media types. I don't think we need to do it anymore.

Reply:
changed it to only pull for pullspec_image